src/main/java/com/atlassian/bitbucket/jenkins/internal/credentials/CredentialUtils.java

```diff
@@ -26,19 +25,12 @@ private CredentialUtils() {
                 CredentialUtils.class.getName() + " should not be instantiated");
     }
 
+    // If the context is null, then the Jenkins context will be used.
     public static Optional<Credentials> getCredentials(@Nullable String credentialsId, @Nullable Item context) {
```

Comment:
My immediate thought reading this is, when doesn't our code have the item context? Credentials are typically converted by our delegates (from CredentialUtils) that have the context injected and I don't think there's _any_ case a nullable Item context should be accepted- even if it's root to jenkins, we should be providing the Jenkins object explicitly, rather than falling into the trap of thinking context is optional in resolving credentials.
Not your code, but I think we should remove the `@Nullable` annotation from both fields here. Null credential ids should be handled by the consumer (they are in all usages of this method already), and I think a misunderstanding of the item context has led to a lot of hidden bugs. Given Jenkins resolves a null context to `Jenkins.get()` anyway, I think that'll make the code more maintainable in the long run.

Reply:
It looks like `BitbucketScmFormFill` allows `@Nullable` items. I think this is information is coming from the UI and we get it from the REST request, so it is nullable.  I don't know if we can guarantee a non-null context in all cases

Reply:
That's strange too. The form fill delegate should always have context from the `@AncestorInPath`- it's injected directly from stapler. Did we add that annotation? None of the injectors in the SCM or the Source are nullable. @dkjellin do you know about this?